### ISSUE TITLE
Regarding how can I load my custom event in misp-feeds

### ISSUE THREAD

**Issue by eagerwolverine** [MISP/MISP#9040]
Hey I just want to know how can I load my custom-event in the feeds section. As we know when we load feed then default events those feeds gets imported on the misp-events.
So please help!

Thank you in advance!

**Comment by Nicolas-Pellletier** [comment]
Hello,
I'm not sure to understand what you mean by "load my custom-event in the feeds section" and the rest of this sentance: "we load feed then default events those feeds gets imported on the misp-events."  It wasn't clear for me as well. 

But if you mean import your own feed in order to be able to load other events than the ones fetched through the default misp feed, you can do it by going to the https://MISP_URL/feeds/index. Then on the left panel you could add your own feed...

**Comment by eagerwolverine** [comment]
Ok let me just make this clear for you...if I created any custom event for my organisation and I want to see its details loaded into list feeds tab how can I do that??

Please help...
Thank you in advance!

**Comment by Nicolas-Pellletier** [comment]
The events created with feed doesn't have any information from which feed is coming from. there is no link between event and feed in the data model. So i think you can't.

**Comment by eagerwolverine** [comment]
So is there an opposite way possible so that first we create a feed then with the help of that feed can we load some events.

Please do let me know.

Thank you in advance.!

**Comment by Nicolas-Pellletier** [comment]
That is exacly what the feed does... See [here](https://www.circl.lu/doc/misp/managing-feeds/)  or [here](https://misp.gitbooks.io/misp-book/content/managing-feeds/) for an explanation of how it works and how to proceed in order to make it works.

**Comment by UFOSmuggler** [comment]
hi @eagerwolverine 

the misp application itself cannot generate feeds, only consume them.  you need to make the feed yourself, host it somewhere, and make it available to consumers.  this isn't really described in detail anywhere (that i am aware of) in the misp documentation.

a misp feed is basically a manifest json file that summarises the events available in the feed, and the event json files that contain the full events referenced by the manifest.  the event files take the form of {uuid}.json, where uuid is the event uuid. 
 when a misp instance tries to fetch your feed it will first look for the manifest.json file first, read what is in there, and then decide which of the event files to fetch.  there is also a hashes.csv file that is sometimes used (though not strictly necessary in my experience) that contains md5s of the attributes of each event.  if you update an event, you must update the corresponding timestamp in the feed manifest to let consumers know that event has changed.

here's a pymisp example of a misp feed generator which handles everything described above: https://github.com/MISP/PyMISP/tree/main/examples/feed-generator

if you just want to submit your event to an existing feed provider, you will have to contact that feed provider directly and build a relationship with them.  they might give you an api key that allows you to push your event into their community.

it's admirable that you want to share your threat intel, and don't let the difficulties involved stop you.  once you work through the rough edges it should become easier.

good luck mate.

**Comment by eagerwolverine** [comment]
So how can I load my custom events on a feed and fetch them through MISP feeds by providing an url on our MISP platform!?

Do I need to host url somewhere in order to fetch them through this way??

**Comment by UFOSmuggler** [comment]
yes, you need to create the feed as the event and manifest files, then host those on whatever url you desire, ie. https://www.example.com/

when you put https://www.example.com/ into your misp instance as a feed, the files must be available like:

https://www.example.com/manifest.json
https://www.example.com/c8708f0f-8459-4ab2-a05a-834250002507.json
etc...

if the url was instead https://www.example.com/misp/, then the files must be available like
https://www.example.com/misp/manifest.json

this will result in a working feed.

**Comment by eagerwolverine** [comment]
And what about storing the misp feeds locally i.e. in which directory I need to save the json file of feed, so that I can load up it into misp feed... Since there is also an option in Add feed that we can setup input source to local.

So just tell me path for local directory currently I'm using misp locally on oracle virtualbox vm. As you can see in the ss attached that I selected the input source as local so what should the url??
![image](https://user-images.githubusercontent.com/83109951/235589016-c1b4c229-1036-44d6-8853-bd1330951d99.png)

**Comment by UFOSmuggler** [comment]
any path should be fine as long as the httpd has read access to that path.
